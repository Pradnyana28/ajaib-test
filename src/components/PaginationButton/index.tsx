import React from 'react';
import {
    Flex,
    IconButton,
    Stack,
    Spacer,
} from '@chakra-ui/react';
import { ChevronLeftIcon, ChevronRightIcon } from '@chakra-ui/icons';
import PaginateButton from '../PaginateButton';

const PaginationButton = (props: {
    page: number;
    totalPage: number;
    handlePreviousPage?: () => void;
    handleNextPage?: () => void;
    handlePaginateButton?: (pageNumber: number) => void
}) => {
    return (
        <Flex minWidth='max-content'>
            <p>Page <b>{props.page}</b> of {props.totalPage}</p>
            <Spacer />
            <Stack spacing={1} direction='row' align='center'>
                <IconButton
                    aria-label='Previous page'
                    isActive={props.page <= 1}
                    size='sm'
                    onClick={props.page > 1 ? props.handlePreviousPage : undefined} icon={<ChevronLeftIcon />}
                />
                <PaginateButton
                    totalPage={props.totalPage}
                    page={props.page}
                    handlePaginateButton={props.handlePaginateButton}
                />
                <IconButton
                    aria-label='Next page'
                    isActive={props.page >= props.totalPage}
                    size='sm' onClick={props.page < props.totalPage ? props.handleNextPage : undefined}
                    icon={<ChevronRightIcon />} />
            </Stack>
        </Flex>
    )
}

export default PaginationButton;