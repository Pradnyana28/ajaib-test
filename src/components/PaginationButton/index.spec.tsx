import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PaginationButton from '.';

describe('<PaginationButton />', () => {
    it('renders without crashing', async () => {
        const { container } = render(<PaginationButton page={1} totalPage={10} />);
        expect(container).toMatchSnapshot();
    });
});