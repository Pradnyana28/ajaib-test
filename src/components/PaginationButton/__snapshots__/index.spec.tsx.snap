// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButton /> renders without crashing 1`] = `
<div>
  <div
    class="css-k634du"
  >
    <p>
      Page 
      <b>
        1
      </b>
       of 
      10
    </p>
    <div
      class="css-17xejub"
    />
    <div
      class="chakra-stack css-193rmy8"
    >
      <button
        aria-label="Previous page"
        class="chakra-button css-p7mkev"
        data-active=""
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        data-active=""
        type="button"
      >
        1
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        2
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        3
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        4
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        5
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        6
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        7
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        8
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        9
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        10
      </button>
      <button
        aria-label="Next page"
        class="chakra-button css-p7mkev"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
